src/target/riscv/riscv-013.c

```diff
@@ -3043,3 +3045,472 @@ void riscv013_clear_abstract_error(struct target *target)
 	/* Clear the error status. */
 	dmi_write(target, DMI_ABSTRACTCS, abstractcs & DMI_ABSTRACTCS_CMDERR);
 }
+
+#define COMPLIANCE_TEST(b, message) \
+{                                   \
+	int pass = 0;		    \
+	if (b) {		    \
+		pass = 1;	    \
+		passed_tests++;     \
+	}			    \
+	LOG_INFO("%s test %d (%s)\n", (pass) ? "PASSED" : "FAILED",  total_tests, message); \
+	assert(pass);		    \
+	total_tests++;		    \
+}
+
+int riscv013_test_compliance(struct target *target)
+{
+	LOG_INFO("Testing Compliance against RISC-V Debug Spec v0.13");
+
+	if (!riscv_rtos_enabled(target)) {
+		LOG_ERROR("Please run with -rtos riscv to run compliance test.");
```

Comment:
It may no longer be necessary, I'll try just removing the check

Reply:
I think the reason I required that -rtos be enabled is because we assume we control all harts in this test. But perhaps those two concerns are orthogonal?

Reply:
The only way to do a proper test is to have control of all harts. That doesn't really fit with the OpenOCD model, but we can more-or-less shoehorn it in. Given that OpenOCD is single-threaded, I don't think you need to do anything to ensure full control, except either inform the user that all harts are now in an unknown state, or try hard to restore their state. (I'd pick the former.)